symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/AbstractDatafeedService.java

```diff
@@ -26,93 +32,91 @@
-                for (RealTimeEventListener listener : listeners) {
-                    eventType.dispatch(listener, event);
-                }
-            } catch (IllegalArgumentException e) {
-                log.warn("Receive events with unknown type: {}", event.getType());
-            }
+  /**
+   * Handle a received listener by using the subscribed {@link RealTimeEventListener}.
+   *
+   * @param events List of Datafeed events to be handled
+   */
+  protected void handleV4EventList(List<V4Event> events) {
+    for (V4Event event : events) {
+      if (event == null || event.getType() == null) {
+        continue;
+      }
+      if (this.isSelfGeneratedEvent(event)) {
+        continue;
+      }
+      try {
+        RealTimeEventType eventType = RealTimeEventType.valueOf(event.getType());
+        for (RealTimeEventListener listener : listeners) {
+          eventType.dispatch(listener, event);
         }
+      } catch (IllegalArgumentException e) {
+        log.warn("Receive events with unknown type: {}", event.getType());
+      }
     }
+  }
 
-    private boolean isSelfGeneratedEvent(V4Event event) {
-        return event.getInitiator() != null && event.getInitiator().getUser() != null
-                && event.getInitiator().getUser().getUsername() != null
-                && event.getInitiator().getUser().getUsername().equals(this.bdkConfig.getBot().getUsername());
-    }
+  private boolean isSelfGeneratedEvent(V4Event event) {
+    return event.getInitiator() != null && event.getInitiator().getUser() != null
+        && event.getInitiator().getUser().getUsername() != null
+        && event.getInitiator().getUser().getUsername().equals(this.bdkConfig.getBot().getUsername());
+  }
+
+  protected Map<Predicate<ApiException>, ConsumerWithThrowable> getSessionRefreshStrategy() {
+    return Collections.singletonMap(ApiException::isUnauthorized, e -> refresh());
+  }
 
-    protected Retry getRetryInstance(String name, RetryConfig... config) {
-        Retry retry = config.length == 0 ? Retry.of(name, this.retryConfig) : Retry.of(name, config[0]);
-        retry.getEventPublisher().onRetry(event -> {
-            long intervalInMillis = event.getWaitInterval().toMillis();
-            double interval = intervalInMillis / 1000.0;
-            if (event.getLastThrowable() != null) {
-              log.debug("Datafeed service failed due to {}", event.getLastThrowable().getMessage());
-            }
-            log.info("Retry in {} secs...", interval);
-        });
-        return retry;
+  protected boolean isNetworkOrServerOrUnauthorizedOrClientError(Throwable t) {
+    if (t instanceof ApiException && t.getSuppressed().length == 0) {
+      ApiException apiException = (ApiException) t;
+      return apiException.isServerError() || apiException.isUnauthorized() || apiException.isClientError();
     }
+    return t instanceof ProcessingException;
+  }
```

Comment:
I fill like we could move so a separate class, as it is not only related to Datafeed